slidechain/slidechain_test.go

```diff
@@ -564,18 +576,25 @@ func isPostPegOutTx(tx *bc.Tx, asset xdr.Asset, amount int64, tempAddr, exporter
 	if err != nil {
 		return false
 	}
-	ref := pegOut{
-		AssetXDR: assetXDR,
-		TempAddr: tempAddr,
-		Seqnum:   seqnum,
-		Exporter: exporter,
-		Amount:   amount,
-		Anchor:   anchor,
-		Pubkey:   pubkey,
-	}
-	refdata, err := json.Marshal(ref)
-	if !bytes.Equal(refdata, tx.Log[2][2].(txvm.Bytes)) {
-		return false
+	if foundRetire {
+		var logRef pegOut
+		err := json.Unmarshal(tx.Log[2][2].(txvm.Bytes), logRef)
```

Comment:
This needs `&logRef`, not `logRef`. As it is, this should be a no-op - `logRef` should remain the zero `pegOut` object after this call. But that raises the question: why does the test calling this function not fail?

Reply:
Great catch! Thanks to looking into this, I realized this logic was, in fact, not being executed. Instead, the "refund" clause was always called. The `State` of the `pegOut p` was not being set correctly to `PegOutOk` before being sent to `doPostPegOut`. As a result, the post-peg-out program always refunded the funds, rather than retiring them. This is fixed now.